biorbd_optim/problem_type.py

```diff
@@ -217,7 +217,12 @@ def configure_q_qdot(nlp, as_states, as_controls):
 
         if as_states:
             nlp["x"] = vertcat(nlp["x"], q, q_dot)
-            nlp["var_states"] = {"q": nlp["nbQ"], "q_dot": nlp["nbQdot"]}
+            if "var_states" in nlp.keys():
```

Comment:
var_states (and var_controls) should always exist in an OCP. So I suggest that you remove this if (and subsequent) and add an empty var_states and var_controls in at the creation of the NLP (OptimalControlProgram.__initialize_nlp)